core/client/app/components/gh-cm-editor.js

```diff
@@ -0,0 +1,52 @@
+/* global CodeMirror */
+import Ember from 'ember';
+
+var CodeMirrorEditor = Ember.Component.extend({
+
+    // DOM stuff
+    name: 'cm-editor', // this is so there is a value, just in case an id isn't specified
+    classNameBindings: ['isFocused:focused'],
+    isFocused: false,
+
+    value: '', // make sure a value exists
+    editor: null, // reference to CodeMirror editor
+    mode: 'htmlmixed',
+    theme: 'base16-light',
+
+    // list of default options
+    options: {
+        lineNumbers: true,
+        indentUnit: 4
+    },
+
+    didInsertElement: function () {
+        var options = this.get('options'),
+            self = this,
+            editor;
+        options.mode = this.get('mode');
+        options.theme = this.get('theme');
+        editor = CodeMirror.fromTextArea(document.getElementById(this.get('name')), options);
```

Comment:
Ah, I see you want the text area inside the component. You can still drop the `name` work and just do `this.$('textarea')` to get the textarea inside this component